lib/Runtime/Language/InterpreterStackFrame.cpp

```diff
@@ -6744,7 +6745,10 @@ const byte * InterpreterStackFrame::OP_ProfiledLoopBodyStart(const byte * ip)
             // mark the stackFrame as 'in try block'
             this->m_flags |= InterpreterStackFrameFlags_WithinTryBlock;
 
-            CacheSp();
+            if (shouldCacheSP)
```

Comment:
>shouldCacheSP [](start = 16, length = 13)

Do we also need flip shouldCacheSP right after this if in case we re-enter ProcessTryFinally?